.github/workflows/pypi-nightly-build.yml

```diff
@@ -41,6 +47,9 @@ jobs:
       - name: Set release version
         run: |
           RELEASE_VERSION=$(date +%Y%m%d)
+          if [ "${{ inputs.build_number }}" -gt 0 ]; then
+            RELEASE_VERSION="${RELEASE_VERSION}0${{ inputs.build_number }}"
```

Comment:
We have tried this before, but unfortunately, pypi will keep the longer release version to be the latest version, ie it will never take the new nightly build with shorter release version again.

Reply:
Ahh, how about we just use `date +%Y%m%d%H%M%S` then?

Reply:
or limit build_number to 1 digit, and does not omit it by default, which would be short and reliable (no TZ issue)